lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1156,12 +1156,21 @@ void addIntelFPGADecorations(
     SPIRVEntry *E,
     std::vector<std::pair<Decoration, std::string>> &Decorations) {
   for (const auto &I : Decorations) {
-    if (I.first == DecorationMemoryINTEL)
+    switch (I.first) {
+    case DecorationMemoryINTEL:
       E->addDecorate(new SPIRVDecorateMemoryINTELAttr(E, I.second));
-    else {
+      break;
+    case DecorationRegisterINTEL:
+    case DecorationSinglepumpINTEL:
+    case DecorationDoublepumpINTEL:
+      assert(I.second.empty());
+      E->addDecorate(I.first);
+      break;
+    default:
```

Comment:
I guess 'default' could also be replaced by a full listing of remaining attributes (clarification for future conributors). @asavonic @AlexeySachkov please reply if this is necessary

Reply:
I'm not sure that it is necessary. Probably, just a comment like `The rest of Intel FPGA-specific decorations (DecorationA, DecorationB, etc.)` could be helpful

Reply:
Thanks, I'll add the comment then